probe_scraper/glean_push.py

```diff
@@ -0,0 +1,181 @@
+    schema, _, bucket_path = output_bucket.partition("://")
+    if schema == "gs":
+        bucket_name, _, prefix = bucket_path.partition("/")
+        prefix = f"{prefix.rstrip('/')}/glean/{repo}/".lstrip("/")
+        bucket = storage.Client().get_bucket(bucket_name)
+        for root, dirnames, filenames in os.walk(repo_out_dir):
+            file_root = Path(root)
+            blob_root = Path(prefix) / os.path.relpath(root, start=repo_out_dir)
+            for filename in filenames:
+                file_path = file_root / filename
+                blob = bucket.blob(str(blob_root / filename))
+                print(f"Uploading {file_path} to gs://{blob.bucket.name}/{blob.name}")
+                blob.upload_from_filename(file_path)
+    elif schema == "s3":
+        return
+        # cloudfront is supposed to automatically gzip objects, but it won't do that
+        # if the object size is > 10 megabytes (https://webmasters.stackexchange.com/a/111734)
+        # which our files sometimes are. to work around this, we'll regzip the contents into a
+        # temporary directory, and upload that with a special content encoding
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            for root, dirnames, filenames in os.walk(out_dir):
+                rel_root = os.path.relpath(root, start=out_dir)
+                for dirname in dirnames:
+                    os.mkdir(os.path.join(tmpdirname, rel_root, dirname))
+                for filename in filenames:
+                    in_filename = os.path.join(root, filename)
+                    out_filename = os.path.join(tmpdirname, rel_root, filename)
+                    with open(in_filename, "rb") as f1:
+                        with gzip.open(out_filename, "wb") as f2:
+                            f2.write(f1.read())
+
+            repo_out_dir = os.path.join(out_dir, glean, repo, "")
+            compressed_repo_out_dir = os.path.join(tmpdirname, glean, repo, "")
+            subprocess.check_call(
+                [
+                    "aws",
+                    "s3",
+                    "sync",
+                    compressed_repo_out_dir,
+                    f"{output_bucket}/glean/{repo}/",
+                    "--delete",
+                    "--content-type",
+                    "application/json",
+                    "--content-encoding",
+                    "gzip",
+                    "--cache-control",
+                    "max-age=28800",
+                    "--acl",
+                    "public-read",
+                ]
+            )
+    else:
+        raise ValueError(
+            f"unsupported output bucket format, must start with gs:// or s3://, but got {output_bucket!r}"
+        )
+
+
+def main(request: Request) -> Response:
+    """Scrape probes from a single glean commit."""
+    output_bucket = os.environ.get(
+        "OUTPUT_BUCKET",
+        # "s3://net-mozaws-prod-us-west-2-data-pitmo",
+        "gs://relud1-probe-scraper/gcf_output",
+    )
```

Comment:
TODO: update this with the gcs bucket provided in [DSRE-684](https://mozilla-hub.atlassian.net/browse/DSRE-684)